typegate/native/src/runtimes/s3.rs

```diff
@@ -156,3 +151,71 @@ fn s3_list(client: S3Client, bucket: &str, path: &str) -> S3Items {
         },
     }
 }
+
+#[deno]
+struct S3Upload {
+    key: String,
+}
+
+#[deno]
+enum S3UploadedFile {
+    Ok { key: String },
+    Err { message: String },
+}
+
+#[deno]
+fn s3_upload(client: S3Client, bucket: &str, key: &str, file: &[u8]) -> S3UploadedFile {
+    let mut bucket = match create_bucket(client, bucket).tap_err(|e| error!("{e:?}")) {
+        Ok(bucket) => bucket,
+        Err(e) => {
+            return S3UploadedFile::Err {
+                message: e.to_string(),
+            }
+        }
+    };
+
+    bucket.set_path_style();
+
+    let temp = RT.block_on(bucket.put_object(key, file));
```

Comment:
Side note: I used the s3 runtime in Rust because the deno one was not complete. Now that we can use the one of node, maybe the easier would be keep it in Deno. This is would avoid to load the entire file into memory (when passing from Deno to Rust, and make use of stream to upload the file).